ddtrace/propagation/http.py

```diff
@@ -483,10 +494,192 @@ def _extract(headers):
+        ``sampling_priority`` = ``s``
+        ``origin`` = ``o``
+        ``_dd.p.`` prefix = ``t.``
+    """
+
+    @staticmethod
+    def _get_traceparent_values(tp):
+        # type: (str) -> Tuple[int, int, int]
+        """If there is no traceparent, or if the traceparent value is invalid raise a ValueError.
+        Otherwise we extract the trace-id, span-id, and sampling priority from the
+        traceparent header.
+        """
+
+        version, trace_id_hex, span_id_hex, trace_flags_hex = tp.split("-")
+        # check version is a valid hexadecimal
+        int(version, 16)
+        # currently 00 is the only version format, but if future versions come up we may need to add changes
+        if version != "00":
+            log.warning("unsupported traceparent version:%r, still attempting to parse", version)
+
+        if len(trace_id_hex) == 32 and len(span_id_hex) == 16 and len(trace_flags_hex) >= 2:
+            trace_id = _hex_id_to_dd_id(trace_id_hex)
+            span_id = _hex_id_to_dd_id(span_id_hex)
+
+            # All 0s are invalid values
+            if trace_id == 0:
+                raise ValueError("0 value for trace_id is invalid")
+            if span_id == 0:
+                raise ValueError("0 value for span_id is invalid")
+
+            trace_flags = _hex_id_to_dd_id(trace_flags_hex)
+            # there's currently only one trace flag, which denotes sampling priority
+            # was set to keep "01" or drop "00"
+            # trace flags is a bit field: https://www.w3.org/TR/trace-context/#trace-flags
+            sampling_priority = trace_flags & 0x1
+
+        else:
+            raise ValueError("W3C traceparent hex length incorrect: %s" % tp)
+
+        return trace_id, span_id, sampling_priority
+
+    @staticmethod
+    def _get_tracestate_values(ts):
+        # type: (str) -> Tuple[Optional[int], Dict[str, str], Optional[str]]
+
+        # tracestate parsing, example: dd=s:2;o:rum;t.dm:-4;t.usr.id:baz64,congo=t61rcWkgMzE
+        dd = None
+        ts_l = ts.split(",")
+        for list_mem in ts_l:
+            if list_mem.startswith("dd="):
+                # cut out dd= before turning into dict
+                list_mem = list_mem[3:]
+                dd = dict(item.split(":") for item in list_mem.split(";"))
+
+        # parse out values
+        if dd:
+            sampling_priority_ts = dd.get("s")
+            if sampling_priority_ts is not None:
+                sampling_priority_ts = int(sampling_priority_ts)
+
+            origin = dd.get("o")
+            # need to convert from t. to _dd.p.
+            other_propagated_tags = {"_dd.p.%s" % k[2:]: v for (k, v) in dd.items() if (k.startswith("t."))}
+
+            return sampling_priority_ts, other_propagated_tags, origin
+        else:
+            return None, {}, None
+
+    @staticmethod
+    def _get_sampling_priority(sampling_priority_tp, sampling_priority_ts):
```

Comment:
Non-blocker: using the names specific to each extracted value could help reading this code.

```suggestion
    def _get_sampling_priority(traceparent_sampled, tracestate_sampling_priority):
```